pwiz_tools/Skyline/Alerts/ShareResultsFilesDlg.cs

```diff
@@ -0,0 +1,452 @@
+                    foreach (var chromFileInfo in chromatogramSet.MSDataFileInfos)
+                    {
+                        // Check for path validity, using our standard rules for locating data files when they aren't in current working directory
+                        if (ScanProvider.FileExists(Program.MainWindow.DocumentFilePath, chromFileInfo.FilePath, out var path))
+                        {
+                            paths.Add(path);
+                        }
+                        else
+                        {
+                            missingPaths.Add(chromFileInfo.FilePath.GetFileName());
+                        }
+                    }
+                }
+                var repFiles = paths.ToList(); // Convert to list. Prevents duplicates from being present
+                repFiles.Sort(NaturalComparer.Compare); // Natural Sort
+                
+                // Add to list view for selection
+                checkedListBox.Items.AddRange(paths.ToArray());
+                SelectOrDeselectAll(true); // All check box elements start selected
+
+                var missingRepFiles = missingPaths.ToList(); // Convert to list. Prevents duplicates from being present
+                missingRepFiles.Sort(NaturalComparer.Compare);
+
+                missingListBox.Items.AddRange(missingRepFiles.ToArray()); //Add all elements present to 
+            }
+        }
+
+
+        /// <summary>
+        /// Accept button pressed. Selection information saved
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Btn_Accept_Click(object sender, EventArgs e)
+        {
+            OkDialog();
+        }
+
+
+        /// <summary>
+        /// Save all list box and checked list box information about the users current
+        /// selection of what to included in the zip file as well as what is still currently
+        /// missing.
+        /// </summary>
+        public void UpdateAuxFiles()
+        {
+            var checkedItems = (from object checkedItem in checkedListBox.Items
+                select checkedListBox.CheckedItems.Contains(checkedItem)
+                    ? new CheckedListBoxItems(checkedItem.ToString(), true)
+                    : new CheckedListBoxItems(checkedItem.ToString(), false)).ToList();
+
+            // Add all missing auxiliary files to list
+            var missingItems = (from object missingItem in missingListBox.Items select missingItem.ToString()).ToList();
+
+            // Save auxiliary file selection 
+            _auxiliaryFiles = new AuxiliaryFiles(checkedItems, missingItems); // Save users file selection
+        }
+
+
+        /// <summary>
+        ///  Close form after updating files
+        /// </summary>
+        public void OkDialog()
+        {
+            // Add auxiliary file data
+            UpdateAuxFiles();
+            this.DialogResult = DialogResult.OK;
+            this.Close();
```

Comment:
Setting the DialogResult of a modal dialog causes it to clause. You probably should not call "Close" here.